lacework/resource_lacework_integration_aws_org_agentless_scanning.go

```diff
@@ -0,0 +1,321 @@
+	},
+	"scan_host_vulnerabilities": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Whether to includes scanning for host vulnerabilities.",
+	},
+	"account_id": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "The aws account id",
+	},
+	"bucket_arn": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "The bucket arn",
+	},
+	"scanning_account": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The scanning aws account id",
+	},
+	"monitored_accounts": {
+		Type:     schema.TypeList,
+		Required: true,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+			StateFunc: func(val interface{}) string {
+				return strings.TrimSpace(val.(string))
+			},
+		},
+		Description: "The list of monitored aws accounts ids or OUs",
+	},
+	"management_account": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "The scanning aws account id",
+	},
+	"credentials": {
+		Type:        schema.TypeList,
+		MaxItems:    1,
+		Optional:    true,
+		Description: "The credentials needed by the integration",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"external_id": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The external id",
+				},
+				"role_arn": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The role arn",
+				},
+			},
+		},
+	},
+	"enabled": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "The state of the external integration.",
+	},
+	"retries": {
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Default:     5,
+		Description: "The number of attempts to create the external integration.",
+	},
+	"created_or_updated_time": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"created_or_updated_by": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"type_name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"org_level": {
+		Type:     schema.TypeBool,
+		Computed: true,
+	},
+	"server_token": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"uri": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+}
+
+func resourceLaceworkIntegrationAwsOrgAgentlessScanningCreate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+		retries  = d.Get("retries").(int)
+	)
+
+	awsOrgAgentlessScanningData := api.AwsSidekickOrgData{
+		ScanFrequency:           d.Get("scan_frequency").(int),
+		ScanContainers:          d.Get("scan_containers").(bool),
+		ScanHostVulnerabilities: d.Get("scan_host_vulnerabilities").(bool),
+		AccountID:               d.Get("account_id").(string),
+		BucketArn:               d.Get("bucket_arn").(string),
+		ScanningAccount:         d.Get("scanning_account").(string),
+		MonitoredAccounts:       strings.Join(d.Get("monitored_accounts").([]string), ", "),
```

Comment:
This cast fails,  we can call the following instead
```suggestion
		MonitoredAccounts:       strings.Join(castAttributeToStringSlice(d, "monitored_accounts"), ", "),
```

Reply:
Good call, TBH I was worried about this, as the proper way is to first try `[]interface{}` then for `string` on each element.